metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -183,6 +197,38 @@ func getCustomFields(customFields []types.BaseCustomFieldValue, customFieldsMap
 	return outputFields
 }
 
+func getNetworkNames(c *vim25.Client, ref types.ManagedObjectReference) ([]string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	outputNetworkNames := []string{}
+
+	pc := property.DefaultCollector(c)
+
+	var vm mo.VirtualMachine
+	err := pc.RetrieveOne(ctx, ref, []string{"network"}, &vm)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving virtual machine information: %v", err)
+	}
+
+	if len(vm.Network) == 0 {
+		return nil, errors.Wrap(err, "no networks found")
+	}
+
+	var nets []mo.Network
+	err = pc.Retrieve(ctx, vm.Network, []string{"name"}, &nets)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving network from virtual machine: %v", err)
+	}
+
+	for _, net := range nets {
+		name := strings.Replace(net.Name, ".", "_", -1)
```

Comment:
We probably should make this a helper function for in beats to "normalise" keys.

Reply:
This would be great!